test/unit/taxonomy/topic_taxonomy_test.rb

```diff
@@ -32,4 +32,14 @@ def subject
                             build(:taxon_hash, content_id: 'visible_id', visibility: false),])
     assert_equal ['visible_id'], subject.visible_taxons.map(&:content_id)
   end
+
+  test "#all taxons are ordered by name" do
+    redis_cache_has_taxons([build(:taxon_hash, title: 'Cow', phase: 'live'),
+                            build(:taxon_hash, title: 'Moose', phase: 'alpha'),
+                            build(:taxon_hash, title: 'Alpha', phase: 'alpha'),
+                            build(:taxon_hash, title: 'Donkey', phase: 'beta'),
+                            build(:taxon_hash, title: 'Zebra', phase: 'alpha'),
+                            build(:taxon_hash, title: 'Anteater', phase: 'beta')])
+    assert_equal %w[Alpha Anteater Cow Donkey Moose Zebra], subject.all_taxons.map(&:name)
```

Comment:
I think you are supposed to be testing the assertion against `subject. return_all_taxons` here! 😉

Reply:
😭yes you are correct